modules/frontend/combiner/common.go

```diff
@@ -129,15 +137,24 @@ func (c *genericCombiner[T]) HTTPFinal() (*http.Response, error) {
 		return nil, err
 	}
 
-	bodyString, err := new(jsonpb.Marshaler).MarshalToString(final)
-	if err != nil {
-		return nil, fmt.Errorf("error marshalling response body: %w", err)
+	var bodyString string
+	if c.httpMarshalingFormat == api.HeaderAcceptProtobuf {
+		buff, err := proto.Marshal(final)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling response body: %w", err)
+		}
+		bodyString = string(buff)
+	} else {
+		bodyString, err = new(jsonpb.Marshaler).MarshalToString(final)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling response body: %w", err)
+		}
 	}
 
 	return &http.Response{
-		StatusCode: c.httpStatusCode,
+		StatusCode: 200,
```

Comment:
Doesn't this change the behaviour for all endpoints?

Reply:
No, only the TraceId handler uses that combiner, and it was returning 200 always. Ultimately the status code in the response can be modified in the handler once the whole pipeline has finished

Reply:
Isn't `genericCombiner` used in other combiners? See https://github.com/grafana/tempo/blob/main/modules/frontend/combiner/search_tag_values.go#L17

Reply:
Yes, but in the form of GRPC combiner:

https://github.com/grafana/tempo/blob/6e9d93aefe7f103dc394c068e8d558ff3e4d40db/modules/frontend/combiner/search_tag_values.go#L41

Which doesn't use the HTTPFinal

Reply:
`HTTPFinal()` is called by `httpCollector` in [`RoundTrip()`](https://github.com/grafana/tempo/blob/main/modules/frontend/pipeline/collector_http.go#L47), which is instantiated in all pipelines ([example](https://github.com/grafana/tempo/blob/main/modules/frontend/metrics_query_range_handler.go#L95))

Reply:
But even more, it doesn't matter because of this line:

https://github.com/grafana/tempo/blob/6e9d93aefe7f103dc394c068e8d558ff3e4d40db/modules/frontend/combiner/common.go#L122

Which is going to return an errored HTTP response if the status code != 200. This is kinda wrong since the queriers may return another code in the 200 range like 204.  

So, it doesn't matter at all because it will always return 200 unless there is an error :P

Reply:
> `HTTPFinal()` is called by `httpCollector` in [`RoundTrip()`](https://github.com/grafana/tempo/blob/main/modules/frontend/pipeline/collector_http.go#L47), which is instantiated in all pipelines ([example](https://github.com/grafana/tempo/blob/main/modules/frontend/metrics_query_range_handler.go#L95))

Super confusing this stuff since there is also a GRPCCollector.
I don't mind to change it again if you want

Reply:
Oh okay, I see now. It's definitely confusing.